### PR TITLE
Fix wrong repeat of negative prompt

### PR DESCRIPTION
**Pull request by hadaev8**
With current codebase this snippet would return error

```
num_images = 2
prompt = ["a photograph of an astronaut riding a horse"] * num_images

images = pipe(prompt, negative_prompt=prompt).images
```

Number of positive and negative prompts already checked in line ```elif batch_size != len(negative_prompt):```
So no reason to repeat it by batch size.

### CODE REVIEW

**Comment by hadaev8** [comment]
Im not sure whats wrong with code quality.

**Comment by Thomas-MMJ** [comment]
> Im not sure whats wrong with code quality.

Have you tried running 

`python -m black source_file_name.py
`
and seeing what changes?

**Comment by shirayu** [comment]
Anther formatting way is to run 

```bash
make style
```

**Comment by pcuenca** [comment]
Hi @hadaev8, thanks a lot! However I think this was already in progress in #1002 (issue #779), which was just missing some tests. Can you please take a look there and see if there's anything missing?

**Comment by hadaev8** [comment]
@pcuenca Yes, I checked and change is same, but I fixed all (at least what I found) pipes.
Is it possible to merge our PRs?

**Comment by hadaev8** [comment]
@Thomas-MMJ
It added whitespace.
```

            removed_text = self.tokenizer.batch_decode(text_input_ids[:, self.tokenizer.model_max_length :])
            removed_text = self.tokenizer.batch_decode(text_input_ids[:, self.tokenizer.model_max_length:])
```
Whitespace was removed by vs code and I considered its fine.
Why whitespace should be here? Or is the problem on the side of code quality check?
Im confused.

**Comment by Thomas-MMJ** [comment]
> Whitespace was removed by vs code and I considered its fine.
> Why whitespace should be here? Or is the problem on the side of code quality check?
> Im confused.

It just happens to be the style standard used for this code base.  There are a variety of essentially arbitrary choices for consistency in style and this happens to be one.

**Comment by patrickvonplaten** [comment]
@pcuenca could you take a look here as it's very similar to your open PR here: https://github.com/huggingface/diffusers/pull/1002

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def __call__(
             uncond_embeddings = self.text_encoder(uncond_input.input_ids.to(self.device))[0]
 
             # duplicate unconditional embeddings for each generation per prompt
-            uncond_embeddings = uncond_embeddings.repeat_interleave(batch_size * num_images_per_prompt, dim=0)
+            uncond_embeddings = uncond_embeddings.repeat_interleave(num_images_per_prompt, dim=0)
```
Comment (pcuenca) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:283]: `repeat_interleave` is not compatible with some devices (`mps`, possibly ONNX). Can we do this the same way as the others?

```suggestion
            seq_len = uncond_embeddings.shape[1]
            uncond_embeddings = uncond_embeddings.repeat(1, num_images_per_prompt, 1)
            uncond_embeddings = uncond_embeddings.view(batch_size * num_images_per_prompt, seq_len, -1)
```

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint_legacy.py [hunk 2]
```diff
@@ -299,7 +299,7 @@ def __call__(
             uncond_embeddings = self.text_encoder(uncond_input.input_ids.to(self.device))[0]
 
             # duplicate unconditional embeddings for each generation per prompt
-            uncond_embeddings = uncond_embeddings.repeat_interleave(batch_size * num_images_per_prompt, dim=0)
+            uncond_embeddings = uncond_embeddings.repeat_interleave(num_images_per_prompt, dim=0)
```
Comment (pcuenca) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint_legacy.py:302]:
```suggestion
            seq_len = uncond_embeddings.shape[1]
            uncond_embeddings = uncond_embeddings.repeat(1, num_images_per_prompt, 1)
            uncond_embeddings = uncond_embeddings.view(batch_size * num_images_per_prompt, seq_len, -1)
```

Same comment as before.

**Review by pcuenca — APPROVED** [review]
Looks good to me! It's the same as the other PR but includes all the pipelines.

The only thing remaining would be to create some tests, as outlined by @patil-suraj here: https://github.com/huggingface/diffusers/pull/1002#pullrequestreview-1157249188.

@hadaev8 will you have time to work on it? Otherwise happy to take it :)

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py (hunk 1 above)
Comment (hadaev8) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:283]: Also i think we dont need view here, just repeat batch axis, right?

Comment (pcuenca) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:283]: Yes, you are right, I blindly copied the lines.

**Comment by pcuenca** [comment]
@hadaev8 For now I'd use something like maybe https://github.com/huggingface/diffusers/blob/main/tests/test_pipelines.py#L265 to exercise a small pipeline with different inputs, and verify that no exceptions are raised and the number of images returned is correct. These tests would include all the cases:
- Single items for positive and negative prompts.
- Single items + num_prompts_per_image
- Lists for both.
- Mismatched cardinality between positive and negative (verify an error is returned).
- Lists + num_prompts_per_image

A better test would be to actually run a full stable diffusion pipeline and verify some slice of the results, but this is somewhat more involved and could be done later in my opinion.

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py (hunk 1 above)
Comment (pcuenca) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:283]: Actually, I just tested locally and we do need to do the reshape.

@hadaev8 We want to merge this for today's release as it's a good fix (we'll do the tests later). I'll make these final changes in your PR if you don't mind and we should be set :)

**Comment by pcuenca** [comment]
I made the final changes here, I hope you don't mind @hadaev8, we want to include this very nice fix in our next release :) I also opened #1117 to create the tests so this PR should be ready to merge now. Thanks a lot @hadaev8!

**Comment by pcuenca** [comment]
Running the fast and slow tests locally.

**Comment by hadaev8** [comment]
@pcuenca 
No, i dont mind.

So about this reshape, if repeat doesnt suit, where should be another function to avoid reshaping. Maybe expand or something.

How do you think, does it make sense for another pr or leave it as it is?

**Comment by anton-l** [comment]
Continued in https://github.com/huggingface/diffusers/pull/1120

**Pull request closed without merging**